sudoku_board.h

```diff
@@ -0,0 +1,40 @@
+#pragma
+#include <vector>
+#include <set>
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+
+enum STATUS_OPTION {WIN, LOSE, PROG};
+
+class SudokuBoard {
+private:
+    int m_size;
+    std::vector<std::vector<int>> m_board_values;
+    std::vector<std::vector<std::set<int>>> m_possible_options;
+
+    std::set<int> get_values_in_cube(int r, int c);
```

Comment:
Here the row and column should also be of unsigned types.

Reply:
Also in many other places in your code, I'm not gonna write on any single one of them you can see for yourself